flight/Modules/VtolPathFollower/vtol_follower_fsm.c

```diff
@@ -250,38 +267,58 @@ void configure_timeout(int32_t s)
  */
 
 //! The currently selected goal FSM
-const static struct vtol_fsm_transition *current_goal;
+const static struct vtol_fsm_state *current_goal;
 //! The current state within the goal fsm
-static enum vtol_fsm_state curr_state;
+static enum vtol_fsm_state_num curr_state;
 
-/**
- * Process any sequence of automatic state transitions
- */
-static void vtol_fsm_process_auto()
+static int vtol_fsm_enter_state(enum vtol_fsm_state_num state)
 {
-	while (current_goal[curr_state].next_state[FSM_EVENT_AUTO]) {
```

Comment:
It factors the state transition into a single function rather than the code being duplicated below (vtol_fsm_enter_state).   It also factors timeout configuration into the same place.